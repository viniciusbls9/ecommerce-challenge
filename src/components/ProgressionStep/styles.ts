import styled from 'styled-components'

export const ProgressWrapper = styled.progress``
