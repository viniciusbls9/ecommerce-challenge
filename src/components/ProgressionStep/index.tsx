import * as S from './styles'

const ProgressionStep = () => (
  <S.ProgressWrapper value="50" max="100">
    <h1>ProgressionStep</h1>
  </S.ProgressWrapper>
)

export default ProgressionStep
