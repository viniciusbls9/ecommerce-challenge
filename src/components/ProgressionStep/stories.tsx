import { Story, Meta } from '@storybook/react/types-6-0'
import ProgressionStep from '.'

export default {
  title: 'ProgressionStep',
  component: ProgressionStep
} as Meta

export const Default: Story = () => <ProgressionStep />
