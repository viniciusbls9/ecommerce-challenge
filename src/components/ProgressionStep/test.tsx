import { screen } from '@testing-library/dom'
import { renderWithTheme } from 'utils/helpers'

import ProgressionStep from '.'

describe('<ProgressionStep />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<ProgressionStep currentStep={1} />)

    expect(screen.getByTestId('progress')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
