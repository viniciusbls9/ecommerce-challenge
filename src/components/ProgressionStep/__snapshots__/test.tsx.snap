// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressionStep /> should render the heading 1`] = `
.c0 {
  width: 100%;
  position: relative;
  margin-bottom: 35px;
}

.c1 {
  width: 100%;
  height: 7px;
  border-radius: 14px;
}

.c1::-webkit-progress-bar {
  background-color: #EAEAEA;
  border-radius: 14px;
}

.c1::-webkit-progress-value {
  background: transparent linear-gradient(180deg,#24C92B 0%,#099F0F 100%) 0% 0% no-repeat padding-box;
  border-radius: 14px;
}

.c2 {
  width: 100%;
  position: absolute;
  top: -10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 36px;
  height: 36px;
  border-radius: 50%;
  background-color: #099F0F;
  color: #FAFAFA;
  font-size: 14px;
  font-weight: bold;
}

.c3:first-child {
  margin-left: 20px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 36px;
  height: 36px;
  border-radius: 50%;
  background-color: #EAEAEA;
  color: #FAFAFA;
  font-size: 14px;
  font-weight: bold;
}

.c4:first-child {
  margin-left: 20px;
}

<div
  class="c0"
>
  <progress
    class="c1"
    data-testid="progress"
    max="100"
    value="10"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      1
    </div>
    <div
      class="c4"
    >
      2
    </div>
    <div
      class="c4"
    >
      3
    </div>
  </div>
</div>
`
